### ISSUE TITLE
CS1734 breaks paramrefs with C# 14 extension blocks

### ISSUE THREAD

**Issue by burnchar** [dotnet/roslyn#81217]
**Version Used**: 

**Steps to Reproduce**:

1. Build an extension block on some value
2. Extend that value and accept other parameters
3. Reference the variable that is extended in the block using <paramref>
4. Define at least one <param> tag in the extension method XML for another parameter (other than the extended one)
```cs
/// <param name="value">Foo</param>
extension(ReadOnlySpan<char> value)
{
	/// <param name="n">Bar.</param>
	/// <param name="delimiter">Baz.</param>
	/// <returns><paramref name="value"/>Bing</returns>
	public ReadOnlySpan<char> GetNthDelimitedItem(int n, ReadOnlySpan<char> delimiter)
	{...}
```

**Diagnostic Id**:

0>ReadOnlySpanExtensions.cs(25,32): Error CS1734 : XML comment on 'ReadOnlySpanExtensions.extension(ReadOnlySpan<char>).GetNthDelimitedItem(int, ReadOnlySpan<char>)' has a paramref tag for 'value', but there is no parameter by that name

**Expected Behavior**:
CS1734 is not flagged; detects that the referenced parameter comes from the extension block.

**Actual Behavior**:
Flags extension block parameters as undocumented
